ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/OpenAPIParameterMapper.java

```diff
@@ -151,33 +155,15 @@ public void getResourceInputs(Components components, SemanticModel semanticModel
     }
 
     /**
-     * Map path parameter data to OAS path parameter.
+     *Map path parameters to OpenAPI specification.
      */
     private void createPathParameters(List<Parameter> parameters, NodeList<Node> pathParams) {
         for (Node param: pathParams) {
             if (param instanceof ResourcePathParameterNode pathParam) {
-                PathParameter pathParameterOAS = new PathParameter();
-                if (pathParam.typeDescriptor().kind() == SyntaxKind.SIMPLE_NAME_REFERENCE) {
-                    SimpleNameReferenceNode queryNode = (SimpleNameReferenceNode) pathParam.typeDescriptor();
-                    TypeSymbol typeSymbol = (TypeSymbol) semanticModel.symbol(queryNode).orElseThrow();
-                    componentMapper.addMapping(typeSymbol);
-                    Schema schema = new Schema();
-                    schema.set$ref(MapperCommonUtils.unescapeIdentifier(queryNode.name().text().trim()));
-                    pathParameterOAS.setSchema(schema);
-                } else {
-                    pathParameterOAS.schema(MapperCommonUtils.getOpenApiSchema(
-                            pathParam.typeDescriptor().toString().trim()));
-                }
-
-                pathParameterOAS.setName(MapperCommonUtils.unescapeIdentifier(pathParam.paramName().get().text()));
-
-                // Check the parameter has doc
-                if (!apidocs.isEmpty() && apidocs.containsKey(pathParam.paramName().get().text().trim())) {
-                    pathParameterOAS.setDescription(apidocs.get(pathParam.paramName().get().text().trim()));
-                }
-                // Set param description
-                pathParameterOAS.setRequired(true);
-                parameters.add(pathParameterOAS);
+                PathParameterMapper pathParameterMapper = new PathParameterMapper(
+                        (PathParameterSymbol) semanticModel.symbol(pathParam).get(), openAPI, apidocs,
```

Comment:
When retrieving the path-parameter symbol what happens if the following code segment returns `null` ? 
```java
 (PathParameterSymbol) semanticModel.symbol(pathParam).get()
```

Reply:
When there's a path parameter available, the symbol cannot be null. Before calling this method I have checked whether the path parameter is available or not..
```ballerina
if (!pathParams.isEmpty()) {
   createPathParameters(parameters, pathParams);
}
```